src/codegen.cpp

```diff
@@ -473,6 +473,18 @@ static AttributeList get_func_attrs(LLVMContext &C)
             None);
 }
 
+static AttributeList get_donotdelete_func_attrs(LLVMContext &C)
+{
+    AttributeSet FnAttrs = AttributeSet::get(C, makeArrayRef({Attribute::get(C, "thunk")}));
+    FnAttrs.addAttribute(C, Attribute::InaccessibleMemOnly);
+    FnAttrs.addAttribute(C, Attribute::WillReturn);
+    FnAttrs.addAttribute(C, Attribute::NoUnwind);
```

Comment:
This is not an AttrBuilder. These calls do no have any effects and will be deleted.
```
/Users/jameson/julia1/src/codegen.cpp:479:5: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
    FnAttrs.addAttribute(C, Attribute::InaccessibleMemOnly);
    ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jameson/julia1/src/codegen.cpp:480:5: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
    FnAttrs.addAttribute(C, Attribute::WillReturn);
    ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~
/Users/jameson/julia1/src/codegen.cpp:481:5: warning: ignoring return value of function declared with 'warn_unused_result' attribute [-Wunused-result]
    FnAttrs.addAttribute(C, Attribute::NoUnwind);
    ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
fixed by https://github.com/JuliaLang/julia/pull/44097